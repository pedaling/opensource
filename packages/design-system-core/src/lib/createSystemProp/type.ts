import type { Theme } from '../types';

export type InterpolationFn = (props: Record<string, any>) => Record<string, any>;

export type SystemPropConfig = {
  property: string;
  styleProperty?: string;
  scale?: keyof Pick<Theme, 'colors' | 'space' | 'opacity'>;
  disabled?: boolean;
  transform?: (value: any) => Record<string, any> | undefined;
};

export type SystemProp = {
  (props: Record<string, any>): Record<string, any>;
  propName: string;
  disabled: boolean;
};
