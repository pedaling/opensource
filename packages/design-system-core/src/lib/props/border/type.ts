import type { ResponsiveValue } from '../../types';

export type BorderProps = {
  borderWidth?: ResponsiveValue<number>;
  borderStyle?: ResponsiveValue<'solid' | 'none'>;
  borderColor?: ResponsiveValue<string>;
  borderTopWidth?: ResponsiveValue<number>;
  borderTopStyle?: ResponsiveValue<'solid' | 'none'>;
  borderTopColor?: ResponsiveValue<string>;
  borderRightWidth?: ResponsiveValue<number>;
  borderRightStyle?: ResponsiveValue<'solid' | 'none'>;
  borderRightColor?: ResponsiveValue<string>;
  borderBottomWidth?: ResponsiveValue<number>;
  borderBottomStyle?: ResponsiveValue<'solid' | 'none'>;
  borderBottomColor?: ResponsiveValue<string>;
  borderLeftWidth?: ResponsiveValue<number>;
  borderLeftStyle?: ResponsiveValue<'solid' | 'none'>;
  borderLeftColor?: ResponsiveValue<string>;
  borderRadius?: ResponsiveValue<number | string>;
  borderTopLeftRadius?: ResponsiveValue<number | string>;
  borderTopRightRadius?: ResponsiveValue<number | string>;
  borderBottomLeftRadius?: ResponsiveValue<number | string>;
  borderBottomRightRadius?: ResponsiveValue<number | string>;
  outlineWidth?: ResponsiveValue<number>;
  outlineStyle?: ResponsiveValue<'solid' | 'none'>;
  outlineColor?: ResponsiveValue<string>;
  outlineOffset?: ResponsiveValue<number>;
};
