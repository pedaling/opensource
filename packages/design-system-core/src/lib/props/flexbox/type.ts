import type { ResponsiveValue } from '../../types';

export type FlexboxProps = {
  flex?: ResponsiveValue<string | number>;
  flexGrow?: ResponsiveValue<string | number>;
  flexShrink?: ResponsiveValue<string | number>;
  flexBasis?: ResponsiveValue<string | number>;
  flexDirection?: ResponsiveValue<'column' | 'column-reverse' | 'row' | 'row-reverse'>;
  flexWrap?: ResponsiveValue<'nowrap' | 'wrap' | 'wrap-reverse'>;
  alignContent?: ResponsiveValue<
    | 'space-around'
    | 'space-between'
    | 'space-evenly'
    | 'stretch'
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'start'
    | 'baseline'
    | 'normal'
  >;
  alignItems?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'baseline'
    | 'normal'
    | 'stretch'
  >;
  alignSelf?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'auto'
    | 'baseline'
    | 'normal'
    | 'stretch'
  >;
  justifyContent?: ResponsiveValue<
    | 'space-around'
    | 'space-between'
    | 'space-evenly'
    | 'stretch'
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'start'
    | 'left'
    | 'normal'
    | 'right'
  >;
  justifyItems?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'baseline'
    | 'left'
    | 'legacy'
    | 'normal'
    | 'right'
    | 'stretch'
  >;
  justifySelf?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'auto'
    | 'baseline'
    | 'left'
    | 'normal'
    | 'right'
    | 'stretch'
  >;
  order?: ResponsiveValue<number>;
};
