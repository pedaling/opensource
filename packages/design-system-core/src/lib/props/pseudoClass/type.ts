import type { ResponsiveValue } from '@class101/design-system-core';
import type { SystemProps } from '../systemProp';

export type PseudoClassProps = {
  pseudoHover?: ResponsiveValue<SystemProps>;
  pseudoFocus?: ResponsiveValue<SystemProps>;
  pseudoActive?: ResponsiveValue<SystemProps>;
};
