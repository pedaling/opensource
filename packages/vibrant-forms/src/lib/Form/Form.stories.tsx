import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { Form } from './Form';

export default {
  title: 'Form',
  component: Form,
} as ComponentMeta<typeof Form>;

export const Basic: ComponentStory<typeof Form> = props => <Form {...props} />;
