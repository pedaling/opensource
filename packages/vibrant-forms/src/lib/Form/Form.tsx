import type { FormEvent } from 'react';
import type { FieldValues as ReactHookFormFieldValues } from 'react-hook-form';
import { FormProvider, useForm } from 'react-hook-form';
import { Box } from '@vibrant-ui/core';
import type { FormProps } from './FormProps';

export const Form = <FieldValues extends ReactHookFormFieldValues>({
  children,
  formControlMethods,
  onSubmit,
  ...restProps
}: FormProps<FieldValues>) => {
  const hookFormMethods = useForm<FieldValues>();
  const methods = formControlMethods ?? hookFormMethods;

  return (
    <FormProvider {...methods}>
      <Box
        onSubmit={(e: FormEvent) => {
          e.stopPropagation();

          return onSubmit ? methods.handleSubmit(onSubmit)(e) : undefined;
        }}
        noValidate={true}
        as="form"
        {...restProps}
      >
        {children}
      </Box>
    </FormProvider>
  );
};
