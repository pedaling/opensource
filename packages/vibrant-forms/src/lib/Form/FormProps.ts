import type { FieldValues as ReactHookFormFieldValues, SubmitHandler, UseFormReturn } from 'react-hook-form';
import type { ReactElementChildren } from '@vibrant-ui/core';

export type FormProps<FieldValues extends ReactHookFormFieldValues> = {
  children?: ReactElementChildren;
  formControlMethods?: UseFormReturn<FieldValues>;
  onSubmit?: SubmitHandler<FieldValues>;
  full?: boolean;
};
