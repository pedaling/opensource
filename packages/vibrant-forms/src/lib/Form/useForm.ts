import { useForm } from 'react-hook-form';
export { useForm };
