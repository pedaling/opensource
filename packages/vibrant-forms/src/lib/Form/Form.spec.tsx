import type { ReactRenderer } from '@vibrant-ui/utils/testing-web';
import { createReactRenderer } from '@vibrant-ui/utils/testing-web';
import { Form } from './Form';

describe('<Form />', () => {
  const { render } = createReactRenderer();

  let renderer: ReactRenderer;

  describe('when Form rendered', () => {
    beforeEach(async () => {
      renderer = render(<Form data-testid="form" />);
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });
});
