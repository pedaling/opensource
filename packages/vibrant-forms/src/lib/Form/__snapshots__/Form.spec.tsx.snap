// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> when Form rendered match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
}

<div>
  <form
    class="emotion-0"
    data-testid="form"
    novalidate=""
  />
</div>
`;
